qa/integration/specs/logstash_to_logstash_spec.rb

```diff
@@ -96,8 +73,12 @@ def wait_for_logstash(service)
         run_logstash_instance(output_config_name, all_config_options) do |upstream_logstash_service|
 
           try(num_retries) do
-            downstream_event_stats = downstream_logstash_service.monitoring_api.event_stats
+            upstream_event_stats = upstream_logstash_service.monitoring_api.event_stats
+            expect(upstream_event_stats).to include({"out" => num_events}), lambda { "expected #{num_events} events to have been sent by upstream" }
```

Comment:
The trouble with this expectation is that it needs to fire in the window between when the pipeline sends its last event and when the pipeline and process have shut down, and as the backoff window gets bigger, the odds of us missing this window increases.

I fear that this will introduce additional instability.

Reply:
🤦 we have generator input, agree with your point.
I will remove this part shortly and keep only the clean-up changes.